### PR TITLE
Refactor random string

### PR DESCRIPTION
**Pull request by ziima**

### CODE REVIEW

File: openid/store/nonce.py [hunk 1]
```diff
@@ -84,6 +85,19 @@ def checkTimestamp(nonce_string, allowed_skew=SKEW, now=None):
...
 
+def make_nonce_salt(length=6):
+    """
+    Generate and return a nonce salt.
+
+    @param length: Length of the generated string.
+    @type length: int
+    @rtype: six.text_type
+    """
+    sys_random = random.SystemRandom()
```
Comment (tpazderka) [inline openid/store/nonce.py:96]: This can raise `NotImplementedError` on some platforms.

File: openid/test/test_consumer.py [hunk 2]
```diff
@@ -1780,7 +1781,7 @@ def setUp(self):
         # base64(btwoc(g ^ xb mod p))
         self.dh_server_public = cryptutil.longToBase64(self.server_dh.public)
 
-        self.secret = cryptutil.randomString(self.session_cls.secret_size)
+        self.secret = os.urandom(self.session_cls.secret_size)
```
Comment (tpazderka) [inline openid/test/test_consumer.py:1783]: This can raise NotImplementedError on some platforms.

**Review by tpazderka — CHANGES_REQUESTED** [review]

File: openid/store/nonce.py (hunk 1 above)
Comment (ziima) [reply openid/store/nonce.py:96]: Yes, it can. But I consider it as the best solution. It seems better than using not-very-random source.

Comment (tpazderka) [reply openid/store/nonce.py:96]: I think that adding a fallback is a reasonable choice :) Also `secrets` module on p3.6 provides a nice interface to cryptography

Comment (ziima) [reply openid/store/nonce.py:96]: `secrets` is just wrap over `SystemRandom`, so it doesn't offer any additional fallback.

I'd rather have error raised, than fallback to not very secure Mersenne Twister.

**Review by tpazderka — APPROVED** [review]
OK, the error behavior is up to you :)

**Pull request merged**